### ISSUE TITLE
Confusing Liskov diagnostics when signature types in subclass have the same name but differ

### ISSUE THREAD

**Issue by adevore** [astral-sh/ty#2687]
```python

# file: a.py

from package_x import SomeType

class ParentClass:
    def foo(arg: SomeType) -> None:
        pass

# file b.py

from package_y import SomeType

class SubClass(ParentClass):
    def foo(arg: SomeType) -> None:
        pass
```

This produces an invalid-method-override error. I encountered it when my IDE incorrectly added an import from the wrong module. The error message is confusing because the methods are visually identical.

This similar Rust snippet:

```Rust
mod foo {
    pub struct Bar;
}

mod baz {
    pub struct Bar;
}

fn function(arg: baz::Bar) {}

use foo::Bar;

fn main() {
    function(Bar);
}
```

Instead produces this error:

```
error[E0308]: mismatched types
  --> src/main.rs:14:14
   |
14 |     function(Bar);
   |     -------- ^^^ expected `baz::Bar`, found `foo::Bar`
   |     |
   |     arguments to this function are incorrect
   |
   = note: `foo::Bar` and `baz::Bar` have similar names, but are actually distinct types
note: `foo::Bar` is defined in module `crate::foo` of the current crate
  --> src/main.rs:2:5
   |
 2 |     pub struct Bar;
   |     ^^^^^^^^^^^^^^
note: `baz::Bar` is defined in module `crate::baz` of the current crate
  --> src/main.rs:6:5
   |
 6 |     pub struct Bar;
   |     ^^^^^^^^^^^^^^
note: function defined here
  --> src/main.rs:9:4
   |
 9 | fn function(arg: baz::Bar) {}
   |    ^^^^^^^^ -------------
```

**Comment by carljm** [comment]
Thanks for the report! We already disambiguate same-named types when they show up in diagnostics, but the problem here is just that we don't provide detailed diagnostics about why two callables are incompatible at all. If we did, I think our existing disambiguation logic would kick in. So I think this is a duplicate of #1644
